spt/OrangeBox/modules/vguimatsurfaceDLL.cpp

```diff
@@ -0,0 +1,449 @@
+			vertIndex += 3;
+		if (cl_showfps && cl_showfps->GetBool())
+			++vertIndex;
+	}
+
+	const int BUFFER_SIZE = 256;
+	wchar_t buffer[BUFFER_SIZE];
+	currentVel = clientDLL.GetPlayerVelocity();
+	Vector accel = currentVel - previousVel;
+	std::string info(y_spt_hud_ent_info.GetString());
+
+	if (!whiteSpacesOnly(info))
+	{
+		int entries = utils::FillInfoArray(info, INFO_ARRAY, MAX_ENTRIES, INFO_BUFFER_SIZE, PROP_SEPARATOR, ENT_SEPARATOR);
+		for (int i = 0; i < entries; ++i)
+		{
+			DrawSingleString(vertIndex, INFO_ARRAY + INFO_BUFFER_SIZE * i, fontTall, BUFFER_SIZE, x, surface, buffer);
+		}
+	}
+
+	if (y_spt_hud_velocity.GetBool())
+	{
+		DRAW_TRIPLEFLOAT(L"vel(xyz)", currentVel.x, currentVel.y, currentVel.z);
+		DRAW_FLOAT(L"vel(xy)", currentVel.Length2D());
+	}
+
+	if (y_spt_hud_accel.GetBool())
+	{
+		DRAW_TRIPLEFLOAT(L"accel(xyz)", accel.x, accel.y, accel.z);
+		DRAW_FLOAT(L"accel(xy)", accel.Length2D());
+	}
+
+	if (y_spt_hud_script_length.GetBool())
+	{
+		swprintf_s(buffer, BUFFER_SIZE, L"frame: %d / %d", scripts::g_TASReader.GetCurrentTick(), scripts::g_TASReader.GetCurrentScriptLength());
+		DRAW();
+	}
+
+#if SSDK2007
+	if (y_spt_hud_portal_bubble.GetBool())
+	{
+		int in_bubble = GetEnviromentPortal() != NULL;
+		DRAW_INT(L"portal bubble", in_bubble);
+	}
+#endif
+
+	if (y_spt_hud_flags.GetBool())
+	{
+		int flags = clientDLL.GetPlayerFlags();
+		DRAW_FLAGS(NULL, FLAGS, flags, false);
+	}
+
+	if (y_spt_hud_moveflags.GetBool())
+	{
+		int flags = serverDLL.GetPlayerMoveType();
+		DRAW_FLAGS(L"Move type", MOVETYPE_FLAGS, flags, true);
+	}
+
+	if (y_spt_hud_collisionflags.GetBool())
+	{
+		int flags = serverDLL.GetPlayerCollisionGroup();
+		DRAW_FLAGS(L"Collision group", COLLISION_GROUPS, flags, true);
+	}
+
+	if (y_spt_hud_movecollideflags.GetBool())
+	{
+		int flags = serverDLL.GetPlayerMoveCollide();
+		DRAW_FLAGS(L"Move collide", MOVECOLLIDE_FLAGS, flags, true);
+	}
+
+	if (y_spt_hud_vars.GetBool() && utils::serverActive())
+	{
+		auto vars = clientDLL.GetMovementVars();
+		DRAW_FLOAT(L"accelerate", vars.Accelerate);
+		DRAW_FLOAT(L"airaccelerate", vars.Airaccelerate);
+		DRAW_FLOAT(L"ent friction", vars.EntFriction);
+		DRAW_FLOAT(L"frametime", vars.Frametime);
+		DRAW_FLOAT(L"friction", vars.Friction);
+		DRAW_FLOAT(L"maxspeed", vars.Maxspeed);
+		DRAW_FLOAT(L"stopspeed", vars.Stopspeed);
+		DRAW_FLOAT(L"wishspeed cap", vars.WishspeedCap);
+		DRAW_INT(L"onground", (int)vars.OnGround);
+	}
+
+#ifndef OE
+	if (y_spt_hud_ag_sg_tester.GetBool() && utils::serverActive())
+	{
+		Vector v = clientDLL.GetPlayerEyePos();
+		QAngle q;
+
+		std::wstring result = calculateWillAGSG(v, q);
+		swprintf_s(buffer, BUFFER_SIZE, L"ag sg: %s", result.c_str());
+		DRAW();
+	}
+#endif
+
+	if (y_spt_hud_oob.GetBool())
+	{
+		Vector v = clientDLL.GetPlayerEyePos();
+		trace_t tr;
+		Strafe::Trace(tr, v, v + Vector(1,1,1), Strafe::HullType::POINT);
+
+		bool oob = engineDLL.ORIG_CEngineTrace__PointOutsideWorld(nullptr, 0, v) && !tr.startsolid;
```

Comment:
This sounds like it'll report inbounds even when part of the player is sticking out into oob.